lib/services/lastfm.js

```diff
@@ -0,0 +1,95 @@
+const axios = require('axios');
+const moment = require('moment');
+
+class LastFm {
+  constructor(config) {
+    this.fetcher = axios.create({
+      baseURL: config.url,
+    });
+    this.apiKey = config.apiKey;
+    this.username = config.username;
+  }
+
+  fetchLastScrobbledTrack(username, limit, page, from, extended, to) {
+    return this.fetcher.get('/', {
+      params: {
+        method: 'user.getRecentTracks',
+        user: username,
+        limit,
+        page,
+        from,
+        extended,
+        to,
+        api_key: this.apiKey,
+        format: 'json',
+      },
+    })
+      .then(response => response.data);
+  }
+
+  getProfilePage() {
+    return new URL(`https://last.fm/user/${this.username}`);
+  }
+
+  getCurrentPlayingSong() {
+    return this.fetchLastScrobbledTrack(this.username, 1).then((body) => {
+      const lastPlayed = body.recenttracks.track[0];
+      if (lastPlayed) {
+        const trackName = lastPlayed.name;
+        const artistName = lastPlayed.artist['#text'];
+        const nowPlaying = lastPlayed['@attr'] && lastPlayed['@attr'].nowplaying === 'true';
+
+        if (nowPlaying) {
+          return {
+            trackName,
+            artistName,
+            nowPlaying,
+          };
+        }
+
+        const lastPlayedTimestamp = moment.unix(lastPlayed.date.uts);
+        const playedAgo = moment.duration(lastPlayedTimestamp.diff(moment()));
+
+        return {
+          trackName,
+          artistName,
+          playedAgo,
+          nowPlaying,
+        };
+      }
+      return null; // `No song has been scrobbled to ${this.getProfilePage()}`;
+    }).catch((err) => {
+      console.log(err);
```

Comment:
Only thing left to do from my perspective here is maybe a unit test with a mocked fetcher to test that objects are parsin' correctly.

And then get rid of these catches so they can be handled by the downstream callers.